pkg/tools/cli/loadtest/service/search.go

```diff
@@ -17,26 +17,61 @@ package service
 
 import (
 	"context"
+	"sync"
 
+	"github.com/vdaas/vald/apis/grpc/agent/core"
 	"github.com/vdaas/vald/apis/grpc/gateway/vald"
 	"github.com/vdaas/vald/apis/grpc/payload"
+	"github.com/vdaas/vald/internal/errors"
 	"github.com/vdaas/vald/internal/net/grpc"
 	"github.com/vdaas/vald/pkg/tools/cli/loadtest/assets"
+	"github.com/vdaas/vald/pkg/tools/cli/loadtest/config"
 )
 
-func newSearch() (requestFunc, loaderFunc) {
-	return func(dataset assets.Dataset) ([]interface{}, error) {
-			vectors := dataset.Query()
-			requests := make([]interface{}, len(vectors))
-			for j, v := range vectors {
-				requests[j] = &payload.Search_Request{
-					Vector: v,
-				}
+func searchRequestProvider(dataset assets.Dataset) (func() interface{}, int, error) {
+	v := dataset.Query()
+	size := len(v)
+	i := 0
+	m := &sync.Mutex{}
+	return func() (ret interface{}) {
+		m.Lock()
+		defer m.Unlock()
+		if i < size {
+			ret = &payload.Search_Request{
+				Vector: v[i],
 			}
-			return requests, nil
-		},
-		func(ctx context.Context, c vald.ValdClient, i interface{}, copts ...grpc.CallOption) error {
-			_, err := c.Search(ctx, i.(*payload.Search_Request), copts...)
-			return err
+			i++
 		}
+		return ret
+	}, size, nil
+}
+
+func (l *loader) newSearch() (loadFunc, error) {
```

Comment:
I think this method should not have in loader:thinking:

https://github.com/vdaas/vald/pull/552/files#r462742394

Reply:
I think that it is better for insert and search to have same interface.